flair/models/sequence_tagger_model.py

```diff
@@ -941,6 +942,91 @@ def _fetch_model(model_name) -> str:
 
         return model_path
 
+    def _generate_model_card(self, repo_id):
+        return f"""---
+tags:
+    - flair
+    - token-classification
+    - sequence-tagger-model
+---
+
+### Demo: How to use in Flair
+
+Requires:
+- **[Flair](https://github.com/flairNLP/flair/)** (`pip install flair`)
+
+```python
+from flair.data import Sentence
+from flair.models import SequenceTagger
+
+# load tagger
+tagger = SequenceTagger.load("{repo_id}")
+
+# make example sentence
+sentence = Sentence("On September 1st George won 1 dollar while watching Game of Thrones.")
+
+# predict NER tags
+tagger.predict(sentence)
+
+# print sentence
+print(sentence)
+
+# print predicted NER spans
+print('The following NER tags are found:')
+
+# iterate over entities and print
+for entity in sentence.get_spans('ner'):
+    print(entity)
+```"""
+
+    def push_to_hub(
+        self,
+        repo_id: str,
+        token: Optional[str] = None,
+        private: bool = None,
+        commit_message: str = "Add new SequenceTagger model.",
+    ):
+        """
+        Uploads the Sequence Tagger model to a Hugging Face Hub repository.
+        :param repo_id: A namespace (user or an organization) and a repo name separated by a `/`.
+        :param token: An authentication token (See https://huggingface.co/settings/token).
+        :param private: Whether the repository is private.
+        :param commit_message: Message to commit while pushing.
+        :return: The url of the repository.
+        """
+        # Lazy import
+        from huggingface_hub import create_repo, upload_folder
+
+        repo_url = create_repo(
+            repo_id=repo_id,
+            token=token,
+            private=private,
+            exist_ok=True,
+        )
+
+        with tempfile.TemporaryDirectory() as tmp_dir:
+            tmp_path = Path(tmp_dir)
+
+            # Save model weight
+            local_model_path = tmp_path / "pytorch_model.bin"
+            self.save(local_model_path)
+
+            # Generate and save model card
+            model_card_content = self._generate_model_card(repo_id)
+            readme_path = tmp_path / "README.md"
+            if not readme_path.exists():
+                with readme_path.open("w", encoding="utf-8") as f:
+                    f.write(model_card_content)
```

Comment:
@osanseviero I would be careful here. If the repo already exists on the Hub (which is possible since `exist_ok=True` is used in `create_repo`) and the user manually updated the README, pushing a new version of `SequenceTagger` would overwrite it without the user noticing it.

To avoid that we need to know if the readme exists on the repo. I don't think we have a proper way in `huggingface_hub` to know that. A workaround is to try to download the readme from the repo (if it exists, I expect 
 the useless download to be small compared to uploading a `SequenceTagger`).

```py
from huggingface_hub import hf_hub_download
from huggingface_hub.utils import EntryNotFoundError

(...)

write_readme = True # whether you should write a readme or not
try:
    hf_hub_download(repo_id=repo_id, filename="README.md", use_auth_token=token)
    write_readme = False
except EntryNotFoundError:
    pass
```

(Just mentioning it to say I don't like it but it's also possible to set `exists_ok` to False and catch a `HTTPError` with status_code HTTP409. I think it's too low level for a third-party library.)

Reply:
Good point! A somewhat cleaner way is to use `model_info` to obtain all files in the repo, and just add one if it's not there.

Reply:
Done in https://github.com/flairNLP/flair/pull/2897/commits/0c2bc3dfcdecde74e4599c5ddec84107898bba4e

Reply:
Even better ! :)